### PR TITLE
Feat: Add a rolling window to circuit breaker strategy

### PR DESCRIPTION
**Pull request by Rcuz8**
**Please describe the changes this PR makes and why it should be merged:**

Right now, there's no way to reset the circuit breaker environment between unit tests, meaning earlier tests will effect later tests.

This PR adds a rolling timeframe to the circuit breaker strategy.

**Status and versioning classification:**

- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: src/strategies/circuit-breaker.strategy.ts [hunk 1]
```diff
@@ -53,6 +57,18 @@ export class CircuitBreakerStrategy extends Strategy<CircuitBreakerOptions> {
...
 
+				if (this.options.rollingWindowInMilliseconds) {
+					const rollingWindowHasExpired = () =>
+						Date.now() >=
+						state.lastRequestTimeMs + this.options.rollingWindowInMilliseconds;
+					if (rollingWindowHasExpired()) {
+						state.failuresCount = 0;
+						state.succeedsCount = 0;
+						state.openedAt = 0;
+						state.lastRequestTimeMs = Date.now();
```
Comment (SocketSomeone) [inline src/strategies/circuit-breaker.strategy.ts:68]: Why we set last request only when rolling window has expired not every request? If it`s has no meaning - it`ll be merged

**Review by SocketSomeone — CHANGES_REQUESTED** [review]
LGTM, but has some questions :)

File: src/strategies/circuit-breaker.strategy.ts (hunk 1 above)
Comment (Rcuz8) [reply src/strategies/circuit-breaker.strategy.ts]: You're right it should be every request. Made the change here: [d309e91](https://github.com/SocketSomeone/nestjs-resilience/pull/320/commits/d309e913775d06f6b67153a76474b21c7cc610ad)

File: test/decorators/use-resilience-circuit.decorator.spec.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { CircuitBreakerStrategy, UseResilience } from '../../src';
```
[the remark this answers is not in the document]
Comment (Rcuz8) [reply test/decorators/use-resilience-circuit.decorator.spec.ts]: Sounds good. Made the change here: https://github.com/SocketSomeone/nestjs-resilience/pull/320/commits/248090845090d37bd52fcba8c08a5b7614b81e3d

**Review by SocketSomeone — APPROVED** [review]

**Comment by SocketSomeone** [comment]
@Rcuz8 Thank you for PR! https://github.com/SocketSomeone/nestjs-resilience/releases/tag/2.1.0 published

**Pull request merged**